File: chroma-league/src/main/java/com/bonepl/chromaleague/hud/parts/resource/ManaBar.java
package com.bonepl.chromaleague.hud.parts.resource;

import com.bonepl.chromaleague.hud.colors.CLColor;
import com.bonepl.chromaleague.hud.parts.ProgressBar;
import com.bonepl.chromaleague.state.GameStateHelper;
import com.bonepl.razersdk.animation.AnimatedFrame;
import com.bonepl.razersdk.animation.Frame;

/**
 * vladimir uses resource type: GNARFURY
 * shyvana uses resource type: DRAGONFURY
 */
public class ManaBar extends AnimatedFrame {
    @Override
    public Frame getFrame() {
        addAnimationFrame(new ProgressBar(ResourceBars.getResourceBarKeys(),
                GameStateHelper.getResourcePercentage(), CLColor.MANA));
        return super.getFrame();
    }
}

File: src/hud/parts/resource/ManaBar.ts
import {AnimatedFrame} from '../../../razer-sdk/animation/AnimatedFrame.js';
import {Frame} from '../../../razer-sdk/animation/Frame.js';
import {CLColor} from '../../colors/CLColor.js';
import {ProgressBar} from '../ProgressBar.js';
import {getResourceBarKeys} from './ResourceBars.js';
import {getResourcePercentage} from '../../../state/GameStateHelper.js';

export class ManaBar extends AnimatedFrame {
    pollFrame(): Frame {
        this.addAnimationFrame(new ProgressBar(getResourceBarKeys(), getResourcePercentage(), CLColor.MANA));
        return super.pollFrame();
    }
}
